src/uct/sm/mm/base/mm_md.c

```diff
@@ -130,33 +130,24 @@ ucs_status_t uct_mm_mem_dereg(uct_md_h md, uct_mem_h memh)
     return UCS_OK;
 }
 
-ucs_status_t uct_mm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
+void uct_mm_md_query(uct_md_h md, uct_md_attr_t *md_attr, int support_alloc)
 {
-    md_attr->cap.flags     = 0;
-    md_attr->cap.flags            |= UCT_MD_FLAG_NEED_RKEY;
+    memset(md_attr, 0, sizeof(*md_attr));
+
+    md_attr->cap.flags            = UCT_MD_FLAG_RKEY_PTR |
```

Comment:
why is the UCT_MD_FLAG_RKEY_PTR flag always set here?

Reply:
because for all 'mm' transports it's possible to have direct access to the peer memory - otherwise the FIFO would not have worked. the check 'attach != NULL' we had before is redundant, we don't check it when attaching remote segment for example.